docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -2388,283 +2486,332 @@ The following error message codes may appear on the app-server log. Use the foll
 
   **Action:**
 
-  TODO
+  Review the plugin dependencies as seen in the plugin's pluginDefinition.json file to see if your Zowe configuration or the plugin can be changed in order to match the requirements. Consult the plugin developer if you believe the plugin was able to run fine on the system architecture, so they can explicitly add support in the future.
+
 
 
+### ZWED0177W
 
-### ZWED0176W":"Failed to load client cert/key pair for Caching Service
+  Unable to load _actionOrRecognizer_ for '_pluginId_' into config
 
   **Reason:**
 
-  TODO
+  The plugin _pluginId_ has an action or recognizer within its package and the plugin install process was trying to copy that into the workspace so it can be used, but encountered an error that prevented this.
 
   **Action:**
 
-  TODO
+  Contact support if the reason cannot be determined.
+
 
 
+### ZWED0178W
 
-### ZWED0177W":"Unable to load %s for '%s' into config
+  Skipping authentication plugin _pluginId_ because it's not HA compatible
 
   **Reason:**
 
-  TODO
+  The server is setup for running in high availability (HA) mode which requires that plugins that have state, in particular authentication plugins, must be HA-compatible or else errors will occur. Therefore, the server skips over loading of this plugin nbecause its pluginDefinition.json did not state it was HA compatible.
 
   **Action:**
 
-  TODO
+  Either the plugin must be updated to support and state its support for HA, or it must be removed, or HA mode disabled. To make a plugin support HA, the conformance program should be reviewed. When HA mode is supported, the plugin can be marked as compatible by setting capabilities.haCompatible=true within its initialization.
 
 
 
-### ZWED0178W":"Skipping authentication plugin %s because it's not HA compatible
+### ZWED0179W
+
+  Unable to retrieve the list of certificate authorities from the keyring=_keyringName_ owner=_username_ Error: _error_
 
   **Reason:**
 
-  TODO
+  The server could not automatically determine the certificate authorities (CA) from the z/OS keyring listed. This may cause the server to be unable to verify certificate chains from other servers or clients causing other errors later.
 
   **Action:**
 
-  TODO
+  Review the error to resolve it and contact support if needed. It's also possible as a workaround to explicitly state the CAs within the keyring that you would like to load, rather than relying upon the server's attempt to automatically find all CAs within the keyring.
 
 
 
-### ZWED0179W":"Unable to retrieve the list of certificate authorities from the keyring=%s owner=%s Error: %s
+### ZWED0001E
+
+  Error: _error_
 
   **Reason:**
 
-  TODO
+  The server is running in cluster mode and the cluster manager has encountered an unexpected error.
 
   **Action:**
 
-  TODO
+  Review the error to resolve it, and contact support if needed.
 
 
 
-### ZWED0001E":"Error: %s
+### ZWED0002E
+
+  Could not stop language manager for types=_languageName_
```

Comment:
_languageName_ in this case looks like an array, so may be better to rename it to _languageNames_

Reply:
thanks